### PR TITLE
enhance the error message, test=develop

### PR DESCRIPTION
**Pull request by jerrywgz**
op + api报错信息增强:
- box_coder
- box_decoder_and_assign
- collect_fpn_proposals
- distribute_fpn_proposals
- multiclass_nms

### CODE REVIEW

File: paddle/fluid/operators/detection/box_coder_op.cu [hunk 1]
```diff
@@ -143,20 +141,26 @@ class BoxCoderCUDAKernel : public framework::OpKernel<T> {
...
+                            "Input 'PriorBoxVar' "
+                            "and attribute 'variance' should not"
+                            "be used at the same time."));
       prior_box_var_data = prior_box_var->data<T>();
       prior_box_var_size = prior_box_var->dims().size();
     }
     if (!(variance.empty())) {
-      PADDLE_ENFORCE(static_cast<int>(variance.size()) == 4,
-                     "Size of attribute 'variance' should be 4");
+      PADDLE_ENFORCE_EQ(static_cast<int>(variance.size()) == 4, true,
```
Comment (chenwhql) [inline paddle/fluid/operators/detection/box_coder_op.cu]: why not `PADDLE_ENFORCE_EQ(static_cast<int>(variance.size()), 4`?

File: paddle/fluid/operators/detection/box_coder_op.h [hunk 2]
```diff
@@ -186,16 +186,22 @@ class BoxCoderKernel : public framework::OpKernel<T> {
     const int axis = context.Attr<int>("axis");
     if (target_box->lod().size()) {
       PADDLE_ENFORCE_EQ(target_box->lod().size(), 1UL,
-                        "Only support 1 level of LoD.");
+                        platform::errors::InvalidArgument(
+                            "Input(TargetBox) of "
+                            "BoxCoder only supports 1 level of LoD."));
```
Comment (chenwhql) [inline paddle/fluid/operators/detection/box_coder_op.h]: Input(TargetBox) of BoxCoder operator? add operator is better? other places are same

File: paddle/fluid/operators/detection/box_decoder_and_assign_op.cc [hunk 3]
```diff
@@ -22,53 +22,92 @@ class BoxDecoderAndAssignOp : public framework::OperatorWithKernel {
...
-      PADDLE_ENFORCE_EQ(prior_box_dims[0], target_box_dims[0],
-                        "The first dim of prior_box and target_box is roi nums "
-                        "and should be same!");
-      PADDLE_ENFORCE_EQ(prior_box_dims[0], box_score_dims[0],
-                        "The first dim of prior_box and box_score is roi nums "
-                        "and should be same!");
+      PADDLE_ENFORCE_EQ(
+          prior_box_dims[0], target_box_dims[0],
+          platform::errors::InvalidArgument(
+              "The first dim of prior_box and "
```
Comment (chenwhql) [inline paddle/fluid/operators/detection/box_decoder_and_assign_op.cc]: `dim` -> `dimension`? use whole english word?

File: paddle/fluid/operators/detection/box_decoder_and_assign_op.cc [hunk 4]
```diff
@@ -22,53 +22,92 @@ class BoxDecoderAndAssignOp : public framework::OperatorWithKernel {
...
+              "target_box is roi nums and should be same. But received dim of "
```
Comment (chenwhql) [inline paddle/fluid/operators/detection/box_decoder_and_assign_op.cc]: `roi`? what does is mean? `nums` -> `numbers`?

File: paddle/fluid/operators/detection/box_decoder_and_assign_op.cc [hunk 5]
```diff
@@ -22,53 +22,92 @@ class BoxDecoderAndAssignOp : public framework::OperatorWithKernel {
...
+              "prior_box is %d, dim of target_box is %d",
+              prior_box_dims[0], target_box_dims[0]));
+      PADDLE_ENFORCE_EQ(
+          prior_box_dims[0], box_score_dims[0],
+          platform::errors::InvalidArgument(
+              "The first dim of prior_box and "
```
Comment (chenwhql) [inline paddle/fluid/operators/detection/box_decoder_and_assign_op.cc]: same as above

File: paddle/fluid/operators/detection/distribute_fpn_proposals_op.h [hunk 6]
```diff
@@ -65,8 +65,10 @@ class DistributeFpnProposalsOpKernel : public framework::OpKernel<T> {
     const int num_level = max_level - min_level + 1;
 
     // check that the fpn_rois is not empty
-    PADDLE_ENFORCE_EQ(fpn_rois->lod().size(), 1UL,
-                      "DistributeFpnProposalsOp need 1 level of LoD");
+    PADDLE_ENFORCE_EQ(
+        fpn_rois->lod().size(), 1UL,
+        platform::errors::InvalidArgument("DistributeFpnProposalsOp need 1 "
+                                          "level of LoD"));
```
Comment (chenwhql) [inline paddle/fluid/operators/detection/distribute_fpn_proposals_op.h]: `need 1 level of LoD` means what? -> `need LoD with only one level`? it also exists above

File: paddle/fluid/operators/detection/multiclass_nms_op.cc [hunk 7]
```diff
@@ -26,45 +26,69 @@ class MultiClassNMSOp : public framework::OperatorWithKernel {
...
     auto score_dims = ctx->GetInputDim("Scores");
     auto score_size = score_dims.size();
 
     if (ctx->IsRuntime()) {
-      PADDLE_ENFORCE(score_size == 2 || score_size == 3,
-                     "The rank of Input(Scores) must be 2 or 3");
+      PADDLE_ENFORCE_EQ(score_size == 2 || score_size == 3, true,
+                        platform::errors::InvalidArgument(
+                            "The rank of Input(Scores) must be 2 or 3"
+                            ". But received score_size = %d",
```
Comment (chenwhql) [inline paddle/fluid/operators/detection/multiclass_nms_op.cc]: `scope_size` -> `rank` is better?

File: paddle/fluid/operators/detection/box_coder_op.cu (hunk 1 above)
Comment (jerrywgz) [reply paddle/fluid/operators/detection/box_coder_op.cu]: Done

File: paddle/fluid/operators/detection/box_coder_op.h (hunk 2 above)
Comment (jerrywgz) [reply paddle/fluid/operators/detection/box_coder_op.h]: Done

File: paddle/fluid/operators/detection/box_decoder_and_assign_op.cc (hunk 3 above)
Comment (jerrywgz) [reply paddle/fluid/operators/detection/box_decoder_and_assign_op.cc]: Done

File: paddle/fluid/operators/detection/box_decoder_and_assign_op.cc (hunk 4 above)
Comment (jerrywgz) [reply paddle/fluid/operators/detection/box_decoder_and_assign_op.cc]: Done

File: paddle/fluid/operators/detection/box_decoder_and_assign_op.cc (hunk 5 above)
Comment (jerrywgz) [reply paddle/fluid/operators/detection/box_decoder_and_assign_op.cc]: Done

File: paddle/fluid/operators/detection/distribute_fpn_proposals_op.h (hunk 6 above)
Comment (jerrywgz) [reply paddle/fluid/operators/detection/distribute_fpn_proposals_op.h]: Done

File: paddle/fluid/operators/detection/multiclass_nms_op.cc (hunk 7 above)
Comment (jerrywgz) [reply paddle/fluid/operators/detection/multiclass_nms_op.cc]: Done

**Review by chenwhql — DISMISSED** [review]
LGTM

**Review by chenwhql — APPROVED** [review]

**Pull request merged**